http.go

```diff
@@ -106,6 +116,45 @@ func InstrumentHandlerFunc(label string, f func(http.ResponseWriter, *http.Reque
 	return InstrumentHandler(label, http.HandlerFunc(f))
 }
 
+// traceSampler is a time-based request sampler for tracing.
+//
+// It allows at most one request to be traced during each time interval.
+type traceSampler struct {
+	interval time.Duration
+	rng      *rand.Rand
```

Comment:
I think rand.Rand is not thread-safe, so it should be guarded by a mutex and therefore it should be declared just below the mu field below.

Reply:
Good point, it is not thread-safe.